protocol/migrations/sqlite/1682070624_add_first_unviewed_message_id.up.sql

```diff
@@ -0,0 +1,17 @@
+ALTER TABLE chats ADD COLUMN first_unviewed_message_id VARCHAR;
+
+UPDATE chats
+SET
+    first_unviewed_message_id = (
+        SELECT id
+        FROM user_messages
+        WHERE
+            local_chat_id = chats.id
+            AND NOT(seen)
+            AND NOT(hide)
+            AND NOT(deleted)
+            AND NOT(deleted_for_me)
+        ORDER BY
+            clock_value ASC
+        LIMIT 1
+    );
```

Comment:
I assume this works also for channels that have 0 messages, but I'm asking just in case to know if you tested.